docs/modules/release-notes/pages/0.27.adoc

```diff
@@ -19,30 +21,329 @@ To get started, follow xref:pkl-cli:index.adoc#installation[Installation].#
+We can also see that the module `projectpackage://example.com/birds@1.0.0#/Bird.pkl` resolves to disk location `\file:///my/birds/Bird.pkl` (we can deduce that `birds` is a xref:language-reference:index.adoc#local-dependencies[local dependency]).
+
+To learn more about this feature, consult https://github.com/apple/pkl-evolution/blob/main/spices/SPICE-0001-import-graph-analyzer-api.adoc[SPICE-0001].
 
 == Noteworthy [small]#🎶#
 
 Ready when you need them.
 
-=== XXX
+=== Colored Output
+
+The Pkl CLI will now emit some messages in color (https://github.com/apple/pkl/pull/552[#552], https://github.com/apple/pkl/pull/746[#746]).
+
+Here is a sneak peak of colored error messages in action.
+
+// TODO add image here
+
+Thanks to https://github.com/thomaspurchas[@thomaspurchas] for contributing the feature!
+
+=== `const local` object members
+
+The `const` modifier can be applied to object members, provided that they are also `local` (https://github.com/apple/pkl/pull/678[#678]).
+
+Currently, regular object members are not allowed to have the `const` modifier. This introduces an artificial pain point. 
+
+=== Pkl CLI Changes
+
+==== New CLI Flags
+
+The following common flag has been added:
+
+`--color`::
+Format messages with ANSI color codes.
+
+==== New command: `pkl analyze imports`
+
+As part of the set of APIs added for <<import-analysis,import analysis>>, a new subcommand has been added called `analyze imports` (https://github.com/apple/pkl/pull/695[#695]).
+
+For details, consult the xref:pkl-cli:index.adoc#command-analyze-imports[CLI documentation].
+
+=== Kotlin/Java code generator improvements
+
+=== Pkl Gradle Plugin Changes
+
+==== New `AnalyzeImportsTask`
+
+A new task, `AnalyzeImportsTask`, is introduced (https://github.com/apple/pkl/pull/695[#695]).
+
+This task is the Gradle analogy to `pkl analyze imports`.
+
+For more details, consult the xref:pkl-gradle:index.adoc#analyze-imports[documentation].
+
+==== Tracked file outputs
+
+Tasks created by the Pkl plugin now declare tracked output files (https://github.com/apple/pkl/pull/403[#403]).
+
+This means that downstream tasks do not need to declare an explicit dependency on the Pkl task.
+
+For example, assuming that `evalPkl` is an `EvalTask`:
+
+.build.gradle.kts
+[source,diff]
+----
+ val myGradleTask by tasks.registering {
+   inputs.files(evalPkl)
+-  dependsOn(evalPkl) // <1>
+ }
+----
+<1> No longer necessary to declare this dependency.
+
+[[transitive-modules-computed-by-default]]
+==== `transitiveModules` computed by default 
+
+The `transitiveModules` property of a task is now computed by building the import graph of the source modules (https://github.com/apple/pkl/pull/695[#695]).
+
+This means that Pkl-related tasks no longer need to declare their set of transitive modules, because the pkl-gradle plugin will compute this automatically.
+
+NOTE: This adds latency to each task. To opt out of this behavior, set `transitiveModules` explicitly.
+
+=== Standard Library Changes
```

Comment:
Do the additions to the evaluator settings (external readers, color from #746) need to be covered here too?